pkg/server/server.go

```diff
@@ -433,6 +433,16 @@ func (s *ProxyServer) readFrontendToChannel(stream client.ProxyService_ProxyServ
 func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer, recvCh <-chan *client.Packet) {
 	klog.V(5).Infoln("start serving frontend stream")
 
+	defer func() {
+		// As the read side of the recvCh channel, we cannot close it.
+		// However readFrontendToChannel() may be blocked writing to the channel,
+		// so we need to consume the channel until it is closed.
+		for range recvCh {
```

Comment:
Is it possible to add unit test coverage?

What is the bad thing this prevents - leaked goroutine? If so I suggest explicit in this comment.

Reply:
Without this, the go-routine that actually receives the packets and queues them up in the channel could be blocked here: https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/682e779546f9ebe97952d2ae892e7fd8027a5521/pkg/server/server.go#L422, which would prevent it from reading an EOF from the `Recv()` call: https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/682e779546f9ebe97952d2ae892e7fd8027a5521/pkg/server/server.go#L400-L401, which is what actually triggers the channel to be closed.

So yes, without this the go-routines could be deadlocked.

Note, this was copied from the agent client, which does the same thing: https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/pkg/agent/client.go#L586-L597

Reply:
I realized that `serverRecvBackend` also needs the same treatment, and added it there.

Reply:
I think its fine but if we're worried we could adding a draining connection metric. Increment when we enter the defer and decrement when we exit. If this gets stuck calling the recvCh then the metric will stay up.

Reply:
> I think its fine but if we're worried we could adding a draining connection metric. Increment when we enter the defer and decrement when we exit. If this gets stuck calling the recvCh then the metric will stay up.

IMO: keep the logs for now, and soon add a proxy-server metric for proxy connections labeled by state, similar to the client metric.